samples/nextjs/src/lib/page-props-factory.ts

```diff
@@ -131,7 +137,13 @@ export class SitecorePagePropsFactory {
           componentModule,
         });
       }
+
+      this.layoutPersonalizationUtils.replacePersonalizableComponentsWithLoaderComponents(
```

Comment:
Have you considered extending LayoutService to supply this "switch components" functionality? For example, if there was a LayoutService (no personalization) and a PersonalizationAwareLayoutService, then we could chose which one to initialize in the factory, and then page props wouldn't have to worry about whether there's personalization. This may allow better separation of concerns.

Reply:
Currently we are going to separate personalization in next.js personalization sample so introducing new factory+layoutservice will not bring much benefits to the code.